### ISSUE TITLE
Logs and Traces Missing in Datadog After Upgrading to Azure Functions Worker v2.0.0

### ISSUE THREAD

**Issue by licoto** [DataDog/datadog-aas-extension#295]
After upgrading our .NET 8 Function applications to use Microsoft.Azure.Functions.Worker, Microsoft.Azure.Functions.Worker.Sdk, and Microsoft.Azure.Functions.Worker.Core version 2.0.0, we noticed that logs and traces from Microsoft.Extensions.Logging no longer appear in Datadog. These applications are configured to use [Agentless Logging with APM](https://docs.datadoghq.com/logs/log_collection/csharp/?tab=serilog#agentless-logging-with-apm).

### Environment

- .NET Version: .NET 8
- Azure Functions Version: v4 ([Isolated worker model](https://learn.microsoft.com/en-us/azure/azure-functions/dotnet-isolated-process-guide))
- Datadog AAS Extension: [v3.6.100](https://github.com/DataDog/datadog-aas-extension/releases/tag/dotnet-v3.6.100)

### Temporary Resolution
To restore functionality, we downgraded to the following versions of the Azure Functions Worker libraries:

- Microsoft.Azure.Functions.Worker v1.24.0
- Microsoft.Azure.Functions.Worker.Sdk v1.18.1
- Microsoft.Azure.Functions.Worker.Core v1.20.0

### Additional Reference
We recently reported a similar issue (https://github.com/DataDog/datadog-aas-extension/issues/292#issue-2664706432), which was related to .NET APIs.

**Comment by bouwkast** [comment]
Hi @licoto it looks like `dd-trace-dotnet` doesn't yet support V2 of `Microsoft.Azure.Functions.Worker.Core` which would explain why you are missing logs/traces with it.

I've created a task internally for us to add support for that new version.

**Comment by licoto** [comment]
@bouwkast Thank you so much for your quick answer. In this case can we leave this issue open to track progress on that support ?

**Comment by bouwkast** [comment]
Yep feel free to leave this issue open, I've opened a PR to get this supported here: https://github.com/DataDog/dd-trace-dotnet/pull/6472

**Comment by bouwkast** [comment]
@licoto We've just released version `3.9.0` of the tracer that includes support for V2 of the Azure Functions Worker -> https://github.com/DataDog/dd-trace-dotnet/releases/tag/v3.9.0

**Comment by licoto** [comment]
@bouwkast thank you so much for your message. I confirm we are able to see logs and traces after this fix.
